### PR TITLE
Add tests for parser util, Main Parser and Edit Application Parser

### PR DESCRIPTION
**Pull request by keanecjy**
- Add commandtest util for application and internship
- Update messages for all commands to show index error.
- Update message in edit application command parser to show message usage instead of no field has been edited
- Add MainParserTest for edit and view
- Add EditApplicationParserTest
- Add ApplicationParserUtil and GeneralParserUtil tests

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/GeneralStringUtil.java [hunk 1]
```diff
@@ -7,5 +7,7 @@
 
     public static final String COMMA_WITH_SPACE = ", ";
     public static final String DASH = "-";
+    public static final String SPACE = " ";
+    public static final String BLANK = " ";
```
Comment (seanjyjy) [inline src/main/java/seedu/address/commons/util/GeneralStringUtil.java]: sry whats the difference between space and blank

File: src/main/java/seedu/address/logic/parser/util/GeneralParserUtil.java [hunk 2]
```diff
@@ -58,14 +59,9 @@ private static boolean isPreambleEmpty(ArgumentMultimap argumentMultimap) {
...
-    public static Index getIndexInPreamble(ArgumentMultimap argumentMultimap, String messageUsage)
-            throws ParseException {
-
-        try {
-            return parseIndex(argumentMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, messageUsage));
-        }
+    public static Index getIndexInPreamble(ArgumentMultimap argumentMultimap) throws ParseException {
+        System.out.println(argumentMultimap.getPreamble());
```
Comment (seanjyjy) [inline src/main/java/seedu/address/logic/parser/util/GeneralParserUtil.java]: Dont forget to remove the Sout

**Review by seanjyjy — COMMENTED** [review]
LGTM just some nits to fix

File: src/main/java/seedu/address/commons/util/GeneralStringUtil.java (hunk 1 above)
Comment (keanecjy) [reply src/main/java/seedu/address/commons/util/GeneralStringUtil.java]: Fixed

File: src/main/java/seedu/address/logic/parser/util/GeneralParserUtil.java (hunk 2 above)
Comment (keanecjy) [reply src/main/java/seedu/address/logic/parser/util/GeneralParserUtil.java]: Fixed in my latest commit.

**Review by seanjyjy — APPROVED** [review]
LGTM

**Pull request merged**